File: Constructorio_Net/src/models/SearchResponseInner.java
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

/**
 * Constructor.io Search Response Inner
 **/
namespace Constructorio_NET
{
    public class SearchResponseInner {

        [JsonPropertyName("facets")]
        private List<FilterFacet> facets { get; set; }

        [JsonPropertyName("groups")]
        private List<FilterGroup> groups { get; set; }

        [JsonPropertyName("results")]
        private List<Result> results { get; set; }

        [JsonPropertyName("total_num_results")]
        private Integer totalNumberOfResults { get; set; }

        [JsonPropertyName("sort_options")]
        private List<FilterSortOption> { get; set; }

        [JsonPropertyName("redirect")]
        private Redirect redirect { get; set; }

    }
}

File: Constructorio_Net/src/models/Search/SearchResponseInner.cs
using System.Collections.Generic;
using Newtonsoft.Json;

/**
 * Constructor.io Search Response Inner
 */
namespace Constructorio_NET
{
    public class SearchResponseInner {
        [JsonProperty("facets")]
        public List<FilterFacet> Facets { get; set; }

        [JsonProperty("groups")]
        public List<FilterGroup> Groups { get; set; }

        [JsonProperty("results")]
        public List<Result> Results { get; set; }

        [JsonProperty("total_num_results")]
        public int TotalNumResults { get; set; }

        [JsonProperty("sort_options")]
        public List<FilterSortOption> SortOptions { get; set; }

        [JsonProperty("redirect")]
        public Redirect Redirect { get; set; }
    }
}